### PR TITLE
fix(Rich Text Editor): emoji and mention plugin

### PR DESCRIPTION
**Pull request by bojangles-m**
Emoji plugin
Fixed emoji inline combobox style

Mentions plugin
Fixed inline mentions markup element style

### CODE REVIEW

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { MarkupElementNodeComponentType, MarkupElementNodeType, MarkupElementProps } from './types';
+
+const MarkupElementNode = ({ attributes, element, nodeProps, children }: MarkupElementProps) => (
+    <span
+        {...attributes}
+        data-slate-value={element.value}
+        data-slate-key={element.key}
+        data-slate-category={element.category}
+        contentEditable={false}
+        data-cy={`mention-${element.value?.replaceAll(' ', '-')}`}
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx:16]: `data-cy` looks like to be a cypress tests selector but you don't use it

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/* eslint-disable react/display-name */
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx:1]: Add it only where needed

File: src/components/RichTextEditor/Plugins/MentionPlugin/helpers/filterItems.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { TComboboxItemBase } from '@udecode/plate';
+
+export const filterItems = (value: string) => (item: TComboboxItemBase) =>
+    item.text.toLowerCase().search(new RegExp(`${value}`)) > -1;
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/helpers/filterItems.ts:6]: Isn't it the same as
```
export const filterItems = (value: string) => (item: TComboboxItemBase) =>
    item.text.toLowerCase().includes(value);
```
?

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/types.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+export type MarkupElementProps = { children: ReactNode } & MentionElementProps<Value>;
+export type MarkupElementNodeType = ({ children, ...props }: MarkupElementProps) => JSX.Element;
+
+export type MarkupElementNodeComponentProps = { category: MentionableCategory } & MarkupElementProps;
+export type MarkupElementNodeComponentType = ({
+    category,
+    children,
+    ...props
+}: MarkupElementNodeComponentProps) => JSX.Element;
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/types.ts:15]:
```suggestion
}: MarkupElementNodeComponentProps) => ReactElement;
```
?

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx (hunk 2 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx:1]: Removed

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx (hunk 1 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.tsx:16]: Removed

File: src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/types.ts (hunk 4 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/types.ts:15]: Updated

File: src/components/RichTextEditor/Plugins/MentionPlugin/helpers/filterItems.ts (hunk 3 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/helpers/filterItems.ts:6]: I dont need the toLowerCase() and I can add flag to Regex, would guess this faster then transform and compare. What do you think?

Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/MentionPlugin/helpers/filterItems.ts]: Then also value needs to be lowercase to work

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**